### PR TITLE
Fixes #14065 : Added DBT Cloud connector

### LINKED ISSUE #14065: Add Operational Metadata (Job Details) in DBT Ingestion
Currently DBT Ingestion only pulls the following data from DBT Cloud based on https://docs.open-metadata.org/v1.2.x/connectors/ingestion/workflows/dbt:
image

I am requesting to a new feature to include DBT Operational Metadata such as i.e. Job Name, Last Job Run Schedule Last Job Run Result. 

Business Use case: This will give more information about the Data Asset like if it is fresh/stale, when will be the next data refresh, result of the last data refresh (which can be added in Alerts/Notifications if the job failed).

Thank you!

### PR DESCRIPTION
**Pull request by SumanMaharana**
### Describe your changes:

Fixes #14065 

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/pipeline/dbtcloud/client.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            logger.warning(f"Unable to get run info :{exc}")
+
+        return None
+
+    def get_model_details(self, job_id: int, run_id: int):
+        """
+        get model details for a job in dbt cloud for lineage
+        """
+        try:
+            client_config: ClientConfig = ClientConfig(
```
Comment (OnkarVO7) [inline ingestion/src/metadata/ingestion/source/pipeline/dbtcloud/client.py:95]: We should initialise the client during `init` itself

File: ingestion/src/metadata/ingestion/source/pipeline/dbtcloud/connection.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    metadata: OpenMetadata,
+    client: DBTCloudClient,
+    service_connection: DBTCloudConnection,
+    automation_workflow: Optional[AutomationWorkflow] = None,
+) -> None:
+    """
+    Test connection. This can be executed either as part
+    of a metadata workflow or during an Automation Workflow
+    """
+    test_fn = {
```
Comment (OnkarVO7) [inline ingestion/src/metadata/ingestion/source/pipeline/dbtcloud/connection.py:46]: Should the graphql endpoint also be tested here?

File: ingestion/src/metadata/ingestion/source/pipeline/dbtcloud/metadata.py [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+        """
+        Get Pipeline Status
+        """
+        try:
+            task_status = [
+                TaskStatus(
+                    name=str(task.id),
+                    executionStatus=STATUS_MAP.get(task.state, StatusType.Pending),
+                    startTime=Timestamp(
+                        datetime_to_ts(
```
Comment (OnkarVO7) [inline ingestion/src/metadata/ingestion/source/pipeline/dbtcloud/metadata.py:267]: Do we need to convert the timestamp here in milliseconds?

**Review by OnkarVO7 — CHANGES_REQUESTED** [review]
Left a few comments and also lets add the side panel documentation

File: ingestion/src/metadata/ingestion/source/pipeline/dbtcloud/metadata.py (hunk 3 above)
Comment (SumanMaharana) [reply ingestion/src/metadata/ingestion/source/pipeline/dbtcloud/metadata.py:267]: `datetime_to_ts` converts a given date to a timestamp as an Int in milliseconds

Comment (sushi30) [reply ingestion/src/metadata/ingestion/source/pipeline/dbtcloud/metadata.py:267]: use [datetime.timestamp](https://docs.python.org/3/library/datetime.html#datetime.datetime.timestamp) and multiply by 1000.

**Review by OnkarVO7 — DISMISSED** [review]

File: openmetadata-ui/src/main/resources/ui/src/assets/img/service-icon-dbt.png
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/assets/img/service-icon-dbt.png:1]: This icon is blurry, can you please work with @Rupchavan to have proper icon

**Review by ShaileshParmar11 — COMMENTED** [review]
overall looks good, just replace the icon and we good to go

File: openmetadata-ui/src/main/resources/ui/src/assets/img/service-icon-dbt.png
Comment (SumanMaharana) [reply openmetadata-ui/src/main/resources/ui/src/assets/img/service-icon-dbt.png:1]: Changed the icon with @Rupchavan given icon

**Review by ShaileshParmar11 — APPROVED** [review]

**Review by OnkarVO7 — APPROVED** [review]

**Pull request merged**